<launch>

    <arg name="thename" default="m100"/><!--无人机名称（每台无人机必须不相同）-->
    <arg name="axis" default="2"/><!--坐标系：1为全局自然坐标系，2为局部实验坐标系-->
    <arg name="index" default="0"/>
    <arg name="name" default="groundStation"/>
    <arg name="num" default="10"/>

    <node
        pkg = "cloud_server_communication"
        type = "groundStationClient"
        name = "$(arg name)_client_node"
        output = "screen"
        respawn = "true">
        <param name="uav_index" type="int" value="$(arg index)"/>
        <param name="uav_name" type="string" value="$(arg name)"/>
        <param name="axis_id" type="int" value="$(arg axis)"/>
        <remap from="vel_list" to="velocity_list"/>
    </node>

     <node
        pkg = "cloud_server_communication"
        type = "subscribeGround"
        name = "$(arg name)_subscribeGround"
        output = "screen"
        respawn = "true">
        <param name="uav_index" type="int" value="$(arg index)"/>
        <param name="uav_name" type="string" value="$(arg name)"/>
        <param name="axis_id" type="int" value="$(arg axis)"/>
        <param name="num_uav" type="int" value="$(arg num)"/>
        <remap from="status" to="UAVs/status"/>
    </node>

</launch>
